libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -208,18 +208,18 @@ void NameAndTypeResolver::warnVariablesNamedLikeInstructions() const
 	for (auto const& instruction: evmasm::c_instructions)
 	{
 		string const instructionName{boost::algorithm::to_lower_copy(instruction.first)};
-		auto declarations = nameFromCurrentScope(instructionName, true);
-		for (Declaration const* const declaration: declarations)
+		for (Declaration const* const declaration: nameFromCurrentScope(instructionName, true))
 		{
 			solAssert(!!declaration, "");
-			if (dynamic_cast<MagicVariableDeclaration const* const>(declaration))
-				// Don't warn the user for what the user did not.
-				continue;
-			m_errorReporter.warning(
-				8261_error,
-				declaration->location(),
-				"Variable is shadowed in inline assembly by an instruction of the same name"
-			);
+			// Only warn about variables, everything else is not reachable from
+			// inline assembly.
+			if (auto const* varDecl = dynamic_cast<VariableDeclaration const*>(declaration))
```

Comment:
Ideally we would have the resolution part as a helper, given there are so many weird rules for variables (see `TypeChecker::visit(InlineAssembly)`)

Reply:
I would specifically not add the resolution rules here - we should warn about more than just exactly those variables that are reachable.

On the other hand: This check was introduced in a version of Solidity when yul was not yet "strict". Now, it is not possible to mistakenly use a variable in function-context and vice-versa. So maybe we could remove the check altogether?

Reply:
I think the `reservedIdentifiers` supposed to solve that problem in the dialect, so perhaps we do not need this check here?

Reply:
```
    function g() public pure {
        uint mload;
        assembly {
        }
    }
```

Ah okay so it is for the above case.

I am not sure we have a good solution for this. This may be one of those cases where #2691 would qualify 😅